src/views/SignupPage.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import toastr from 'toastr';
+import SignupForm from '../components/SignupForm';
+import userSignupAction from '../actions/signupActions';
+
+class SignupPage extends Component {
+  constructor(props) {
+    super(props);
+    this.InitialState = {
+      username: '',
+      firstName: '',
+      lastName: '',
+      password: '',
+      confirmPassword: '',
+      email: '',
+    };
+    this.state = { ...this.InitialState, isRequestSent: false };
+    this.isRequestSent = false;
+    this.handleOnChange = this.handleOnChange.bind(this);
+    this.handleOnSubmit = this.handleOnSubmit.bind(this);
+  }
+
+  handleOnChange(event) {
+    this.setState({ [event.target.id]: event.target.value });
+  }
+
+  handleOnSubmit(event) {
+    event.preventDefault();
+    const {
+      username, email, firstName, lastName, password, confirmPassword
+    } = this.state;
+    const { userSignup } = this.props;
+    if (confirmPassword !== password) {
+      return toastr.error('Passwords must match', { timeOut: 3000 });
+    }
+    this.setState({ isRequestSent: true });
+    userSignup({
+      username, email, firstName, lastName, password
+    }).then((response) => {
+      if (response.success) {
+        toastr.success(response.message, { timeOut: 4000 });
+        return this.setState({ ...this.InitialState, isRequestSent: false });
+      }
+      this.setState({ isRequestSent: false });
+      return toastr.error(response.message, { timeOut: 3000 });
```

Comment:
Please let's have a toastr util and a global timeout.
```
showToastr(type, message) {
  return toastr[type](message, { timeOut: 3000 });
}
```
Usage
```
showToastr(success, response.message)
```

Reply:
I have implemented this feedback. Thank you.